azavea/raster-foundry
# Add Airflow Dag Trigger Endpoint

## Overview

This commit makes a few changes to the way Airflow jobs are managed and kicked off.

Airflow DAGs are kicked off in the following situations:

Uploads
 - User changes the status of an upload to `UPLOADED`
 - User creates an upload with status `UPLOADED`
Ingest
 - Scene transitions to state `TOBEINGESTED`
Export
 - When export is created

This PR also brings up the Airflow webserver with the rest of the web stack. The resources are not particularly heavy and this allows user interactions to at least queue up tasks appropriately even if they are not handled by airflow until the airflow stack is brought up.

### Checklist


### Demo

Removed polling dags (except for AOI)
![less-dags](https://cloud.githubusercontent.com/assets/898060/25858998/38bb7eba-34ab-11e7-94a7-9c9927ef3662.png)

Dag Run ID indicates triggered from API
![check-out-dag-name](https://cloud.githubusercontent.com/assets/898060/25859023/42b2c02c-34ab-11e7-9949-0f1356cac5eb.png)

## Testing Instructions

 * Start up the server, when finished and API ready, bring up airflow workers
 * Upload a tif, check logs that task was kicked off properly
 * Add a scene to a project, check ingest task was kicked off
 * Export a project, check that export task was kicked off

Closes https://github.com/azavea/raster-foundry/issues/1695, https://github.com/azavea/raster-foundry/issues/1696
